src/index.ts

```diff
@@ -25,11 +31,13 @@ class Bot {
    * const bot = new Bot()
    */
   public constructor() {
-    this._workingDir = randomTempDir()
-    this._service = new Service(this._workingDir)
-    this.chat = new ChatClient(this._workingDir)
-    this.wallet = new WalletClient(this._workingDir)
-    this.team = new TeamClient(this._workingDir)
+    this._botId = crypto.randomBytes(16).toString('hex')
+    this._workingDir = path.join(os.tmpdir(), `keybase_bot_${this._botId}`)
+    this._service = new Service(this, this._workingDir)
+    this.chat = new ChatClient(this, this._workingDir)
```

Comment:
I'm not sure I'm a big fan of passing a reference of the entire bot to each of the subclients – it creates circular references and starts to break down the distinction between the roles of the clients and the core bot. Could we instead just pass the `debugLog` file path or function to the clients?

Reply:
funny, I started that way, and IMO it gets uglier to go the other way, as things start piling on (bot's working directory, bot's id, bot's log function, etc.)...what's wrong with a circular reference? this seems like the kind of place one makes sense.

Reply:
I understand a lot of the issues with how people use circular references in shitty ways, but these sub-bot clients are 1-1 with their Bot class, and in fact are basically part of the same instance.

hmm, will think about this more.

Reply:
you know...ok...gonna do it your way for now. I may come back to this if more things get tacked on.

Reply:
Ok. I think it'd definitely be worth reevaluating the bot architecture sometime, especially since it's grown a lot since it was first designed.

Reply:
ok, changed it up, lmk. I'd like to merge this soon.